ec2-asg-dynamic-target-tracking/asg-target-setter/app.py

```diff
@@ -0,0 +1,117 @@
+
+# Import modules
+import boto3
+import os
+from pprint import pprint
+import logging
+from botocore.exceptions import ClientError
+from botocore.config import Config
+from datetime import datetime
+from datetime import timedelta
+
+# Create logger
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger()
+
+# Define config
+config = Config(
+   retries = {
+      'max_attempts': 10,
+      'mode': 'standard'
+   }
+)
+
+# Define session and resources
+session = boto3.Session()
+# sqs = session.resource('sqs', config=config)
+cloudwatch = session.client('cloudwatch', config=config)
+autoscaling = session.client('autoscaling', config=config)
+
+# Read environment variables
+asgName = os.environ['asgName']
+desiredLatency = int(os.environ['desiredLatency'])
+defaultMsgProcDuration = int(os.environ['defaultMsgProcDuration'])
+
+# Initialize variables
+metricNamespace = "ASG-Metrics"
+metricName = "MsgProcessingDuration"
+
+
+def publishMetricValue(namespace, metricName, value):
+
+    response = cloudwatch.put_metric_data(
+        Namespace = namespace,
+        MetricData = [
+            {
+                'MetricName': metricName,
+                'Value': value,
+                'StorageResolution': 1
+            }
+        ]
+    )
+
+
+def getMetricValue(metricNamespace, metricName):
+
+    # Define query
+    query = {
+        'Id': 'query_123',
+        'MetricStat': {
+            'Metric': {
+                'Namespace': metricNamespace,
+                'MetricName': metricName,
+            },
+            'Period': 1,
+            'Stat': 'Average',
+        }
+    }
+
+    response = cloudwatch.get_metric_data(
+        MetricDataQueries=[query],
+        StartTime=datetime.utcnow() - timedelta(seconds=86400),
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Naive datetime objects are treated by many datetime methods as local times, We recommend that you create aware timestamps with the `now` or `fromtimestamp` functions using the `tzinfo` to attribute to represent times in UTC. Do not use the [datetime.utcnow()](https://docs.python.org/3/library/datetime.html#datetime.datetime.utcnow) because it returns only a `naive` datetime object. 

[Learn more](https://docs.python.org/3/library/datetime.html#aware-and-naive-objects)

Similar issue at line number 72.